cmd/cloudflared/tunnel/cmd.go

```diff
@@ -1003,6 +1003,13 @@ func configureProxyDNSFlags(shouldHide bool) []cli.Flag {
 			EnvVars: []string{"TUNNEL_DNS_UPSTREAM"},
 			Hidden:  shouldHide,
 		}),
+		altsrc.NewIntFlag(&cli.IntFlag{
+			Name:    tunneldns.MaxUpstreamConnsFlag,
```

Comment:
You can't use `max-upstream-conns` as flag name here, it's not clear that it's used for tunneling DNS only. You need to prefix it with `proxy-dns-`. You'll need to change the code that reads it as well.

Reply:
Good spot, will fix, thanks!